src/coresyftools/coresyftools/tool.py

```diff
@@ -177,3 +178,33 @@ def invoke_shell_command(self, fmt, **kwargs):
         stderr_capture = Capture()
         pipeline = run(cmd_str, stdout=stdout_capture, stderr=stderr_capture)
         return (pipeline, stdout_capture, stderr_capture)
+
+    def prepare_outputs(self):
+        """
+        Method that checks if the tools' outputs are in a directory.
+        If yes, it calls another method to compress. If not, it does nothing.
+        """
+        for out_arg in self.arg_parser.outputs:
+            outputs = self.bindings[out_arg]
+            if not hasattr(outputs, '__iter__'):
+                outputs = [outputs]
+            for output in outputs:
+                if os.path.isdir(output):
+                    self.write_to_zipfile(output)
+
+    def write_to_zipfile(self, directory):
+        """
+        Creates a ZipFile with the files in 'directory'.
+        """
+        path = os.path.abspath(directory)
+        # we need to change the name of the output directory to avoid conflicts
+        # with the zip archive name without the '.zip' extension
+        output_path = path + '_'
+        os.rename(path, output_path)
+        files_list = os.listdir(output_path)
+        with zipfile.ZipFile(path, 'w') as archive:
+            for file_ in files_list:
+                archive.write(os.path.join(output_path, file_), file_)
+            archive.close()
+
```

Comment:
Shouldn't we remove the output directory, `output_path`, after archiving it?

Reply:
Done! The method now removes the output folder after archiving it.